lib/eslint.js

```diff
@@ -718,11 +720,20 @@ module.exports = (function() {
 
                 severity = getRuleSeverity(config.rules[key]);
                 options = getRuleOptions(config.rules[key]);
+                parserName = config.parser;
+
+                // for use against pre-parsed AST, may be no valid parser configured
+                try {
+                    parser = require(parserName);
```

Comment:
BTW, additional benefit of binding `parse` and defining as `RuleContext#parse` is: parsing inside plugins always handles options _exactly_ the same way the host ESLint version does, regardless of what (2.0+) version is installed.

Trying to emulate this inside the plugin when interacting with a parser module directly has been a constant source of confusion (and issues).

So, barring defined use cases for the whole parser module (or even the name), I'd propose simplifying to having a `parse(text)` => AST function on RuleContext.